src/webgpu/util/texture/texel_data.ts

```diff
@@ -505,6 +512,71 @@ export const kTexelRepresentationInfo: {
       },
       pack: components => packComponents([TexelComponent.Stencil], components, 8, 'uint'),
     },
+    'depth24unorm-stencil8': {
+      componentOrder: [TexelComponent.Depth, TexelComponent.Stencil],
+      componentInfo: {
+        Depth: {
+          dataType: 'unorm',
+          bitLength: 24,
+        },
+        Stencil: {
+          dataType: 'uint',
+          bitLength: 8,
+        },
+      },
+      encode: components => {
+        assert(components.Depth !== undefined);
+        assert(components.Stencil !== undefined);
+        assertInIntegerRange(components.Depth, 24, false);
+        assertInIntegerRange(components.Stencil, 8, false);
+        return components;
+      },
+      decode: components => {
+        assert(components.Depth !== undefined);
+        assert(components.Stencil !== undefined);
+        assertInIntegerRange(components.Depth, 24, false);
+        assertInIntegerRange(components.Stencil, 8, false);
+        return components;
+      },
+      pack: components =>
+        packComponents(
+          [TexelComponent.Depth, TexelComponent.Stencil],
+          components,
+          {
+            Depth: 24,
+            Stencil: 8,
+          },
+          'uint'
+        ),
+    },
+    'depth32float-stencil8': {
+      componentOrder: [TexelComponent.Depth, TexelComponent.Stencil],
+      componentInfo: {
+        Depth: {
+          dataType: 'float',
+          bitLength: 32,
+        },
+        Stencil: {
+          dataType: 'uint',
+          bitLength: 8,
+        },
+      },
+      encode: components => {
+        assert(components.Depth !== undefined);
+        assert(components.Stencil !== undefined);
+        assertInIntegerRange(components.Depth, 32, false);
+        assertInIntegerRange(components.Stencil, 8, false);
+        return components;
+      },
+      decode: components => {
+        assert(components.Depth !== undefined);
+        assert(components.Stencil !== undefined);
+        assertInIntegerRange(components.Depth, 32, false);
+        assertInIntegerRange(components.Stencil, 8, false);
+        return components;
+      },
+      pack: () => unreachable('depth32float-stencil8 data cannot be packed'),
```

Comment:
Can depth32float-stencil8 be packed? I said it can't, but I am not 100% sure about it. I am not familiar with the pack / copy related stuff.

Reply:
`pack` is for uploading into a texture. But you can't upload both aspects at once, so we should probably leave this `unreachable` for now

Reply:
it doesn't make sense to pack, as we don't allow copies of both aspects at the same time. In hardware they may be separate planes (8 bytes) or packed (5 bytes) but we hide that.

In fact, this is probably true of `depth24unorm-stencil8` as well. I'm pretty sure we don't allow copies of both aspects at the same time. So it would be ok to make that `pack` also `unreachable()`.

Reply:
Ack.